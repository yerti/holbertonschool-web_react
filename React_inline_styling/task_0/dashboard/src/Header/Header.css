.App-header {
    display: flex;
    align-items: center;
  }
  .App-header img {
    width: 220px;
  }
  .App-header h1 {
    font-size: 40px;
    color: var(--red);
  }