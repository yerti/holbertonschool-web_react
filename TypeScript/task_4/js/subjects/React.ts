namespace Subjects {
  export class React extends Subject {
    experienceTeachingReact?: number;

    getRequirements(): string {
      return super.getRequirements() + "\nJavaScript library: React";
    }
  }
}
