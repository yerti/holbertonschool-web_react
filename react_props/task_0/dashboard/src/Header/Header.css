.App-header {
    display: flex;
    align-items: center;
    border-bottom: 4px solid var(--red);
  }
  .App-header img {
    width: 220px;
  }
  .App-header h1 {
    font-size: 40px;
    color: var(--red);
  }