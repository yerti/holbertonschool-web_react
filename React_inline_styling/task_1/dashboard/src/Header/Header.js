import React from 'react';
import Logo from '../assets/holberton-logo.jpg';
import { StyleSheet, css } from 'aphrodite';

const styles = StyleSheet.create({
  appHeader: {
    display: 'flex',
    alignItems: 'center',
  },
  appHeaderImg: {
    width: '220px',
  },
  appHeaderH1: {
    fontSize: '40px',
    color: 'var(--red)',
  }
})

export default function Header() {
  return (
    <>
      <div className={css(styles.appHeader)}>
        <img className={css(styles.appHeaderImg)} src={Logo} alt='logo of holberton school' />
        <h1 className={css(styles.appHeaderH1)}>School dashboard</h1>
      </div>
    </>
  )
}
