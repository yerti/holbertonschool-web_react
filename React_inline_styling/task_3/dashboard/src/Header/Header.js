import React from 'react';
import Logo from '../assets/holberton-logo.jpg';
import { StyleSheet, css } from 'aphrodite';

const styles = StyleSheet.create({
  appHeader: {
    display: 'flex',
    alignItems: 'center',
  },
  appHeaderImg: {
    width: '220px',
  },
  appHeaderH1: {
    fontSize: '40px',
    color: 'var(--red)',
  },


  smallHeaderTitle: {
    '@media (max-width: 900px)': {
      fontSize: '25px'
    }
  },
  smallImage: {
    '@media (max-width: 900px)': {
      width: '100px'
    }
  },
})

export default function Header() {
  return (
    <>
      <div className={css(styles.appHeader, styles.smallContentHeader)}>
        <img className={css(styles.appHeaderImg, styles.smallImage)} src={Logo} alt='logo of holberton school' />
        <h1 className={css(styles.appHeaderH1, styles.smallHeaderTitle)}>School dashboard</h1>
      </div>
    </>
  )
}
