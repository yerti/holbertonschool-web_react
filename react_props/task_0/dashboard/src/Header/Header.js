import React from 'react'
import Logo from '../assets/holberton-logo.jpg'
import './Header.css'

export default function Header() {
  return (
    <>
      <div className='App-header'>
        <img src={Logo} alt='logo of holberton school' />
        <h1>School dashboard</h1>
      </div>
    </>
  )
}
