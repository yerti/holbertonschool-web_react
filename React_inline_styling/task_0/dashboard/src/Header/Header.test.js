import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";


describe('The Header component', () => {
    it('the Header component to verify it renders without crashing', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper).toMatchSnapshot();
    });

    it('render img and h1 tags', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.find('img')).toHaveLength(1);
        expect(wrapper.find('h1')).toHaveLength(1)
    });

});
