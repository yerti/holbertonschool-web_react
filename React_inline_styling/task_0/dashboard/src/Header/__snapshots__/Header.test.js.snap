// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component the Header component to verify it renders without crashing 1`] = `ShallowWrapper {}`;
